crates/re_query/src/archetype_view.rs

```diff
@@ -322,24 +322,49 @@ impl<A: Archetype> ArchetypeView<A> {
         let component = self.components.get(&C::name());
 
         if let Some(component) = component {
-            let primary_instance_key_iter = self.iter_instance_keys();
+            // NOTE(1): Autogenerated instance keys are interned behind datacells.
+            // If two or more rows in the store share the same keys, then they will share
+            // also the same cells.
+            // Therefore we can compare those cells, and early out if they match.
+            //
+            // NOTE(2): Comparing cells that point to the same backing storage is a simple pointer
```

Comment:
So in case they are not properly interned, this can become a costly comparison, but that cost should be won back by having a faster iterator